test/framework/src/main/java/org/elasticsearch/test/TransportVersionUtils.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.test;
+
+import org.elasticsearch.TransportVersion;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class TransportVersionUtils {
+    private static final List<TransportVersion> ALL_VERSIONS;
+
+    static {
+        ALL_VERSIONS = getDeclaredVersions();
+    }
+
+    /** Returns a random {@link TransportVersion} from all available versions. */
+    public static TransportVersion randomVersion() {
+        return ESTestCase.randomFrom(ALL_VERSIONS);
+    }
+
+    /** Returns a random {@link TransportVersion} from all available versions without the ignore set */
+    public static TransportVersion randomVersion(Set<TransportVersion> ignore) {
+        return ESTestCase.randomFrom(ALL_VERSIONS.stream().filter(v -> ignore.contains(v) == false).collect(Collectors.toList()));
+    }
+
+    /**
+     * Extracts a sorted list of declared transport version constants from a class.
+     */
+    public static List<TransportVersion> getDeclaredVersions() {
```

Comment:
This is the same logic in `TransportVersion` static constructor. Could we have a static method that is used by both? Or just reference `TransportVersion.idToVersion` directly?